src/main/java/picard/illumina/parser/NewIlluminaDataProvider.java

```diff
@@ -66,7 +69,13 @@
         //barcodes
         final Pattern barcodeRegex = Pattern.compile(ParameterizedFileUtil.escapePeriods(
                 ParameterizedFileUtil.makeBarcodeRegex(lane)));
-        final File[] barcodeFiles = getTiledFiles(basecallDirectory, barcodeRegex);
+
+        final File[] barcodeFiles = getTiledFiles(barcodesDirectory, barcodeRegex);
+        if (barcodeFiles.length == 0) {
```

Comment:
Probably should use StringUtils.isEmpty here, since (it looks like) directory.listFiles could return null.

Reply:
Actually that would be ArrayUtils.  Up to you.

Reply:
Probably easier to check `nonNull` vs converting File to String.